book-content/chapters/12-the-weird-parts.md

```diff
@@ -0,0 +1,1653 @@
+
+When you consider this, the empty object type `{}` is a rather elegant solution to the problem of representing anything that isn't `null` or `undefined`.
+
+## The Type and Value Worlds
+
+For the most part, TypeScript can be separated into two syntatical spaces: the type world and the value world. These two worlds can live side-by-side in the same line of code:
+
+```tsx
+const myNumber: number = 42;
+//    ^^^^^^^^  ^^^^^^   ^^
+//    value     type     value
+```
+
+This can be confusing, especially because TypeScript likes to reuse the same keywords across both worlds:
+
+```tsx
+if (typeof key === "string" && (key as keyof typeof obj)) {
+  //^^^^^^^^^^^^^^^^^^^^^^          ^^^^^^^^^^^^^^^^^^^
+  //value                           type
+}
+```
+
+But TypeScript treats this boundary very strictly. For instance, you can't use a type in the value world:
+
+```ts twoslash
+// @errors: 2693
+const processAlbum = (album: Album) => console.log(album);
+// ---cut---
+type Album = {
+  title: string;
+  artist: string;
+};
+
+processAlbum(Album);
+```
+
+As you can see, `Album` doesn't even exist in the value world, so TypeScript shows an error when we try to use it as one.
+
+Another common example is trying to pass a value directly to a type:
+
+```ts twoslash
+// @errors: 2749
+const processAlbum = (album: Album) => console.log(album);
+
+// ---cut---
+type Album = ReturnType<processAlbum>;
+```
+
+In this case, TypeScript suggests using `typeof processAlbum` instead of `processAlbum` to fix the error.
+
+These boundaries are very clear - except in a few cases. Some entities can exist in both the type and value worlds.
+
+### Classes
+
+Consider this `Song` class that uses the shortcut of declaring properties in the constructor:
+
+```tsx
+class Song {
+  title: string;
+  artist: string;
+
+  constructor(title: string, artist: string) {
+    this.title = title;
+    this.artist = artist;
+  }
+}
+```
+
+We can use the `Song` class as a type, for instance to type a function's parameter:
+
+```tsx
+const playSong = (song: Song) =>
+  console.log(`Playing ${song.title} by ${song.artist}`);
+```
+
+This type refers to an _instance_ of the `Song` class, not the class itself:
+
+```ts twoslash
+// @errors: 2345
+class Song {
+  title: string;
+  artist: string;
+
+  constructor(title: string, artist: string) {
+    this.title = title;
+    this.artist = artist;
+  }
+}
+
+const playSong = (song: Song) =>
+  console.log(`Playing ${song.title} by ${song.artist}`);
+
+// ---cut---
+const song1 = new Song("Song 1", "Artist 1");
+
+playSong(song1);
+
+playSong(Song);
+```
+
+In this case, TypeScript shows an error when we try to pass the `Song` class itself to the `playSong` function. This is because `Song` is a class, and not an instance of the class.
+
+So, classes exists in both the type and value worlds, and represents an instance of the class when used as a type.
```

Comment:
but u can get the type of a class with `typeof Cls` if you truly care about the static side of the class